src/lib/update-alternatives/model/alternative.rb

```diff
@@ -101,11 +104,22 @@ def choice(new_choice_path)
       end
       @value = new_choice_path
       @status = "manual"
+      @modified = true
     end
 
     def automatic_mode
       @status = "auto"
       @value = choices.sort_by { |choice| choice.priority.to_i }.last.path
+      @modified = true
+    end
+
+    def save
+      return unless @modified
+      if @status == "auto"
+        UpdateAlternatives::AutomaticModeCommand.new(self).execute
+      else
+        UpdateAlternatives::SetChoiceCommand.new(self).execute
+      end
```

Comment:
btw this way if new command is needed, you just add it to map and everything worked. If you forget to add it and use it in code, then it raise exception so it is easy to fix. Easier then debugging why set choice is called instead of new command.